src/extension/background-script/actions/setup/setIcon.ts

```diff
@@ -0,0 +1,59 @@
+import browser, { Runtime } from "webextension-polyfill";
+import { MessageSetIcon } from "~/types";
+
+import state from "../../state";
+
+const tabIcons = new Map<number, string>();
```

Comment:
can we add a comment what we store here? 

the number is always unique per tab?
And the data stays there even if the tab is closed, correct? (but not a problem because it is not much?)

Reply:
> can we add a comment what we store here?

Sure, will do! 👍

According to the docs the tabId is unique, yes:

➡️ https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/Tab

> The tab's ID. Tab IDs are unique within a browser session. The tab ID may also be set to [tabs.TAB_ID_NONE](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/TAB_ID_NONE) for browser windows that don't host content tabs (for example, devtools windows).

>  And the data stays there even if the tab is closed, correct? (but not a problem because it is not much?)

That's a good point. We could add a `onRemoved` listener to clean up this map after tabs have been closed. I don't think it's a problem (as you said it's not much data being stored in there) but it would feel a bit cleaner. Let me know if I should add it...

Reply:
no code is better than no code... so I think it is fine for now. less event handler are also better I think